src/pod/babashka/aws/impl/aws/credentials.clj

```diff
@@ -0,0 +1,168 @@
+(ns pod.babashka.aws.impl.aws.credentials
+  (:require
+   [clojure.data.json :as json]
+   [clojure.edn]
+   [clojure.java.io :as io]
+   [clojure.java.shell :as shell]
+   [clojure.tools.logging :as log]
+   [cognitect.aws.client.api :as aws]
+   [cognitect.aws.config :as config]
+   [cognitect.aws.credentials :as creds]
+   [cognitect.aws.util :as u]
+   [pod.babashka.aws.impl.aws]))
+
+;;; Pod Backend
+
+ (def *providers (atom {}))
+
+(defn create-provider [provider]
+  (let [provider-id (java.util.UUID/randomUUID)]
+    (swap! *providers assoc provider-id provider)
+    {:provider-id provider-id}))
+
+(defmacro with-system-properties [props & body]
+  `(let [props# (System/getProperties)]
+     (try
+       (doseq [[k# v#] ~props]
+         (System/setProperty k# v#))
+       ~@body
+       (finally
+         (System/setProperties props#)))))
+
+(defn get-provider [config]
+  (get @*providers (get config :provider-id)))
+
+(defn -basic-credentials-provider [conf]
+  (create-provider (creds/basic-credentials-provider conf)))
+
+(defn -environment-credentials-provider []
+  (create-provider (creds/environment-credentials-provider)))
+
+(defn -system-property-credentials-provider [jvm-props]
+  (with-system-properties jvm-props
+    (create-provider (creds/system-property-credentials-provider))))
+
+(defn -profile-credentials-provider
+  ([jvm-props]
+   (with-system-properties jvm-props
+     (create-provider (creds/profile-credentials-provider))))
+
+  ([jvm-props profile-name]
+   (with-system-properties jvm-props
+     (create-provider (creds/profile-credentials-provider profile-name))))
+
+  ([jvm-props profile-name ^java.io.File f]
+   (throw (ex-info "profile-credentials-provider with 2 arguments not supported yet" {}))))
+
+(defn run-credential-process-cmd [cmd]
+  (let [{:keys [exit out err]} (shell/sh "bash" "-c" cmd)]
```

Comment:
This doesn't seem to be cross-platform compatible. I think sending the entire command vector (without bash in it) might be a better approach?

Reply:
I'll fix this post-merge using the below logic.